src/xdr/Stellar-transaction.x

```diff
@@ -236,20 +237,9 @@ struct ChangeTrustOp
 struct AllowTrustOp
 {
     AccountID trustor;
-    union switch (AssetType type)
-    {
-    // ASSET_TYPE_NATIVE is not allowed
-    case ASSET_TYPE_CREDIT_ALPHANUM4:
-        AssetCode4 assetCode4;
-
-    case ASSET_TYPE_CREDIT_ALPHANUM12:
-        AssetCode12 assetCode12;
-
-        // add other asset types here in the future
-    }
-    asset;
+    AssetCode asset;
 
-    // 0, or any bitwise combination of TrustLineFlags
+    // 0, or any bitwise combination of the AUTHORIZED_* flags of TrustLineFlags
```

Comment:
Fixed

Reply:
any reason this isn't an enum then?

Reply:
Good question. `authorize` was initially a `bool`, where `true == AUTHORIZED_FLAG`. For backwards compatibility, it was made a `uint32` in [CAP-0018](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0018.md#specification) where `AUTHORIZED_FLAG == 1` and `AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG == 2`. Using a `uint32` also left the option open to set a combination of multiple flags in the future, which is specifically why an enum wasn't used instead.

I should mention that this operation (`AllowTrustOp`) has been deprecated in favor of `SetTrustLineFlagsOp`, in case you're looking into using these operations.